source/adios2/helper/adiosSystem.cpp

```diff
@@ -105,5 +111,47 @@ int ExceptionToError(const std::string &function)
     }
 }
 
+bool IsHDF5File(const std::string &name, helper::Comm &comm,
+                std::vector<Params> &transportsParameters) noexcept
+{
+    bool isHDF5 = false;
+    if (!comm.Rank())
+    {
+        try
+        {
+            transportman::TransportMan tm(Comm::Duplicate(MPI_COMM_SELF),
+                                          false);
+            if (transportsParameters.empty())
+            {
+                std::vector<Params> defaultTransportParameters(1);
+                defaultTransportParameters[0]["transport"] = "File";
+                tm.OpenFiles({name}, adios2::Mode::Read,
+                             defaultTransportParameters, false);
+            }
+            else
+            {
+                tm.OpenFiles({name}, adios2::Mode::Read, transportsParameters,
+                             false);
+            }
+            const unsigned char HDF5Header[8] = {137, 72, 68, 70,
+                                                 13,  10, 26, 10};
+            if (tm.GetFileSize(0) >= 8)
+            {
+                char header[8];
+                tm.ReadFile(header, 8, 0);
+                tm.CloseFiles();
+                isHDF5 = !std::memcmp(header, HDF5Header, 8);
+            }
+        }
+        catch (std::exception &e)
```

Comment:
Should only be catching exceptions we know about here so we don't swallow and hide ones we can't deal with.

Reply:
I don't want to deal with any error here, any error just means I don't consider this to be a HDF5 file. Let the selected engine (BP3) deal with any error later.

Reply:
By catching the exception you are dealing with it.  So this is essentially dealing with all errors, and explicitly doing nothing with them.  It should instead catch the exceptions it we know to occur and do nothing with them, letting all other exceptions propagate and not get swallowed so they can be dealt with accordingly by upstream callers.

Reply:
Okay, I just catch the file system errors arising from the posix transport operations.